frontend/src/components/accountPage/UserList/UserList.tsx

```diff
@@ -0,0 +1,340 @@
+import { FC, useState } from 'react';
+import {
+  Table,
+  Button,
+  Upload,
+  Modal,
+  Row,
+  Typography,
+  Input,
+  Space,
+} from 'antd';
+import {
+  UploadOutlined,
+  PlusOutlined,
+  SearchOutlined,
+} from '@ant-design/icons';
+import Column from 'antd/lib/table/Column';
+import Papa from 'papaparse';
+import UserListForm from '../UserListForm';
+const Text = Typography.Text;
+
+export type User = {
+  key: Number;
+  userid: string;
+  name: string;
+  surname: string;
+  email: string;
+};
+
+export interface IUserListProps {
+  onAddUser: (users: User[]) => void;
+}
+
+const UserList: FC<IUserListProps> = props => {
+  const [showUserListModal, setshowUserListModal] = useState(false);
+  const [users, setUsers] = useState<User[]>([]);
+  const [fileError, setFileError] = useState<string>('');
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [searchText, setSearchText] = useState<string>('');
+
+  const closeModal = () => setshowUserListModal(false);
+
+  const onCsvUploaded = (fileInfo: any) => {
+    if (fileInfo.file.status === 'removed') {
+      setUsers([]);
+      return;
+    }
+
+    Papa.parse<any>(fileInfo.file, {
+      header: true,
+      skipEmptyLines: true,
+      complete: (result, _) => {
+        for (const line of result.data) {
+          if (
+            !line['NOME'] ||
+            !(line['COGNOME - (*) Inserito dal docente'] || line['COGNOME']) ||
+            !line['MATRICOLA'] ||
+            !line['EMAIL']
+          ) {
+            setFileError(
+              'Invalid file format, must contain <MATRICOLA, NOME, COGNOME (o COGNOME - (*) Inserito dal docente), EMAIL>'
+            );
+            return;
+          }
+        }
+        setUsers(() => {
+          const users = result.data.map((user: any, index: Number) => {
+            return {
+              name: user['NOME'] ?? '',
+              surname:
+                user['COGNOME - (*) Inserito dal docente'] ??
+                user['COGNOME'] ??
+                '',
+              userid: user['MATRICOLA'] ?? '',
+              email: user['EMAIL'] ?? '',
+              key: index,
+            };
+          });
+          props.onAddUser(users);
+          return users;
+        });
+
+        setFileError('');
+      },
+    });
+  };
+
+  return (
+    <>
+      <Table dataSource={users}>
+        <Column
+          title="User ID"
+          dataIndex="userid"
+          sorter={(a: User, b: User) => a.userid.localeCompare(b.userid)}
+          key="userid"
+          filterIcon={<SearchOutlined />}
+          onFilter={(value: any, record: any) =>
+            record['userid']
+              ? record['userid']
+                  .toString()
+                  .toLowerCase()
+                  .includes(value.toLowerCase())
+              : ''
+          }
```

Comment:
Please avoid replicating the code. There are several duplicated lambdas that can be generalized.
In this case you could do something like:
```ts
function makeFilter(field: string) {
  return (val: any, record: any) => 
    record[field] ? record[field].toString().toLowerCase().includes(val.toLowerCase()) : '';
}
```
then do `onFilter={makeFilter('userid')}`
Note: makeFilter is a function that returns a function.